polkadot/node/core/dispute-coordinator/src/scraping/tests.rs

```diff
@@ -662,3 +662,251 @@ fn inclusions_per_candidate_properly_adds_and_prunes() {
+	let candidate_receipt = make_candidate_receipt(get_magic_candidate_hash());
+	let candidate_hash = candidate_receipt.hash();
+
+	// Insert the candidate with multiple block numbers and block hashes
+	let block_numbers = [0, 1, 2];
+	let block_hashes: Vec<_> =
+		block_numbers.iter().map(|&num| get_block_number_hash(num)).collect();
+
+	for (&block_number, &block_hash) in block_numbers.iter().zip(&block_hashes) {
+		inclusions.insert(candidate_hash, block_number, block_hash);
+	}
+
+	// Call the get method for that candidate
+	let result = inclusions.get(&candidate_hash);
+
+	// Verify that the method returns the correct list of block numbers and hashes associated with
+	// that candidate
+	assert_eq!(
+		result.len(),
+		block_numbers.len(),
+		"Expected the same number of results as inserted block numbers"
+	);
+
+	for (&block_number, &block_hash) in block_numbers.iter().zip(&block_hashes) {
+		assert!(
+			result.contains(&(block_number, block_hash)),
+			"Expected to find ({}, {}) in the result",
+			block_number,
+			block_hash
+		);
+	}
+}
+
+#[test]
+fn inclusions_iterative_removal() {
+	let mut inclusions = Inclusions::new();
+
+	// Insert 3 candidates in the specified blocks
+	let candidate1 = make_candidate_receipt(BlakeTwo256::hash(&"c1".encode())).hash();
+	let candidate2 = make_candidate_receipt(BlakeTwo256::hash(&"c2".encode())).hash();
+	let candidate3 = make_candidate_receipt(BlakeTwo256::hash(&"c3".encode())).hash();
+
+	inclusions.insert(candidate1, 0, get_block_number_hash(0));
+	inclusions.insert(candidate2, 1, get_block_number_hash(1));
+	inclusions.insert(candidate3, 0, get_block_number_hash(0));
+	inclusions.insert(candidate3, 2, get_block_number_hash(2));
+
+	// Remove a height that has no blocks and observe no changes
+	inclusions.remove_up_to_height(&0);
+	assert!(inclusions.contains(&candidate1), "Expected candidate1 to remain");
+	assert!(inclusions.contains(&candidate2), "Expected candidate2 to remain");
+	assert!(inclusions.contains(&candidate3), "Expected candidate3 to remain");
+
+	// Remove block 0 and check
+	inclusions.remove_up_to_height(&1);
+	assert!(!inclusions.contains(&candidate1), "Expected candidate1 to be removed");
+	assert!(inclusions.contains(&candidate2), "Expected candidate2 to remain");
+	assert!(inclusions.contains(&candidate3), "Expected candidate3 to remain");
+
+	// Remove block 1 and check
+	inclusions.remove_up_to_height(&2);
+	assert!(!inclusions.contains(&candidate1), "Expected candidate1 to be removed");
+	assert!(!inclusions.contains(&candidate2), "Expected candidate2 to be removed");
+	assert!(inclusions.contains(&candidate3), "Expected candidate3 to remain");
+
+	// Remove block 2 and check
+	inclusions.remove_up_to_height(&3);
+	assert!(!inclusions.contains(&candidate1), "Expected candidate1 to be removed");
+	assert!(!inclusions.contains(&candidate2), "Expected candidate2 to be removed");
+	assert!(!inclusions.contains(&candidate3), "Expected candidate3 to be removed");
+}
```

Comment:
Very good one! 

Another nit for additional check: Add candidate at height 3 and 4. Delete at height 5 and make sure that the candidates at height 3 and 4 are gone.

Reply:
added